### PR TITLE
backupccl: retry restore's AdminSplit on ephemeral errors

### PR DESCRIPTION
**Pull request by msbutler**
Informs #84635,#84162

Release note: none

### CODE REVIEW

File: pkg/ccl/backupccl/split_and_scatter_processor.go [hunk 1]
```diff
@@ -95,13 +98,46 @@ func (s dbSplitAndScatterer) split(
...
+		MaxRetries: 5,
+	}
+
+	var err error
+	for r := retry.StartWithCtx(ctx, retryOpts); r.Next(); {
+
+		if err == nil {
+			break
+		}
+		err := s.db.AdminSplit(ctx, newSplitKey, expirationTime)
```
Comment (adityamaru) [inline pkg/ccl/backupccl/split_and_scatter_processor.go:123]: Is it worth moving this retry logic one level deeper? Maybe we want to ask KV but don't we have a lot of callers outside restore that use `db.AdminSplit` as well? import, index backfills etc.

Comment (adityamaru) [reply pkg/ccl/backupccl/split_and_scatter_processor.go:123]: Or atleast auditing all the places we call `db.AdminSplit` and deciding whether to switch them to this wrapper method.

File: pkg/ccl/backupccl/split_and_scatter_processor.go [hunk 2]
```diff
@@ -95,13 +98,46 @@ func (s dbSplitAndScatterer) split(
...
+
+		if !(kvserver.IsRetriableReplicationChangeError(err)) {
```
Comment (rhu713) [inline pkg/ccl/backupccl/split_and_scatter_processor.go:125]: We might need to add `IsLeaseTransferRejectedBecauseTargetMayNeedSnapshotError()` here or inside some lower level e.g. `IsRetriableReplicationChangeError()` as this error is what #84891 is failing with.

**Review by rhu713 — COMMENTED** [review]
Thanks for the fix!

File: pkg/ccl/backupccl/split_and_scatter_processor.go (hunk 1 above)
Comment (msbutler) [reply pkg/ccl/backupccl/split_and_scatter_processor.go:123]: Good point. I don't see why the caller needs to handle these ephemeral retries. Lemme ask kv what they think in slack.

**Comment by msbutler** [comment]
Closing, as kvserver will handle this internally https://github.com/cockroachdb/cockroach/pull/85405/commits/db2d653cc31d33b0077ad4bd99284ad469a6b4f5

**Pull request closed without merging**